actors/evm/tests/misc.rs

```diff
@@ -0,0 +1,174 @@
+mod asm;
+mod util;
+
+use cid::Cid;
+use evm::interpreter::U256;
+use fil_actor_evm as evm;
+use fvm_ipld_encoding::RawBytes;
+use fvm_shared::econ::TokenAmount;
+
+#[test]
+fn test_timestamp() {
+    let contract = asm::new_contract(
+        "timestamp",
+        "",
+        r#"
+timestamp
+push1 0x00
+mstore
+push1 0x20
+push1 0x00
+return
+"#,
+    )
+    .unwrap();
+
+    let mut rt = util::construct_and_verify(contract);
+    rt.tipset_timestamp = 123;
+    let result = util::invoke_contract(&mut rt, RawBytes::default());
+    assert_eq!(U256::from_big_endian(&result), U256::from(123));
+}
+
+#[test]
+fn test_blockhash() {
+    let contract = asm::new_contract(
+        "timestamp",
+        "",
+        r#"
+push1 0x00
+blockhash
+push1 0x00
+mstore
+push1 0x20
+push1 0x00
+return
+"#,
+    )
+    .unwrap();
+
+    let mut rt = util::construct_and_verify(contract);
+
+    let test_cid =
+        Cid::try_from("bafy2bzacecu7n7wbtogznrtuuvf73dsz7wasgyneqasksdblxupnyovmtwxxu").unwrap();
+    rt.tipset_cids = vec![test_cid];
+    let result = util::invoke_contract(&mut rt, RawBytes::default());
+    assert_eq!(result.to_vec(), test_cid.hash().digest());
```

Comment:
Should we test the lookback case?

Reply:
we could, but I was interesting on the opcode working; not difficult to do either way, will do it if you think it makes sense.